singularity-eos/eos/singularity_eos.cpp

```diff
@@ -230,6 +231,20 @@ static const std::map<const int, const unsigned long> EAPInputToBD = {
     {1, thermalqs::specific_internal_energy | thermalqs::density},
 };
 
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+namespace singularity {
+struct EOSAccessor_ {
+  PORTABLE_INLINE_FUNCTION
+  EOSAccessor_(const Kokkos::View<EOS *, Llft> &eos_v, int *mats)
+    : eos_v_(eos_v), mats_(mats) {}
+  PORTABLE_INLINE_FUNCTION
+  EOS &operator[](const int m) const { return eos_v_(mats_[m]); }
+  Kokkos::View<EOS *, Llft> eos_v_;
+  int *mats_;
+};
+#endif // PORTABILITY_STRATEGY_KOKKOS
```

Comment:
Ah this let's you use the mat array to index into the EOS's?

Reply:
Yep that's correct.